frontend/src/Settings/settingsReducer.js

```diff
@@ -0,0 +1,33 @@
+import {
+  FETCH_SETTINGS_REQUEST, FETCH_SETTINGS_SUCCESS, FETCH_SETTINGS_FAILURE,
+} from './actionCreator';
+
+const initialState = {
+  email: undefined,
+  username: undefined,
+  kingdomName: undefined,
+
+  isLoading: false,
+  error: undefined,
+};
+
+export default function settings(state = initialState, action) {
+  switch (action.type) {
+    case FETCH_SETTINGS_REQUEST:
+      return { ...state, isLoading: true };
+    case FETCH_SETTINGS_SUCCESS:
+      return {
+        ...state,
+
```

Comment:
I don't know why you want to leave a blank line here, just be curious about that.